server/controllers/genAiController.ts

```diff
@@ -0,0 +1,61 @@
+import createAI from '@/server/utils/AIFactory';
+import { createError } from '@/server/middleware/errorHandling';
+
+import type { Request, Response, NextFunction } from 'express';
+import { type GenerativeAIModel } from '@server/types/index';
+import { AIProvider } from '@server/types/index';
+
+const genAIController = {
+  /**
+   * Generate a response from the AI.
+   * The response is stored in res.locals.response
+   *
+   * @param req The request object
+   * @param res The response object
+   * @param next The next function
+   * @return void
+   */
+  async generateResponse(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void> {
+    const AI: AIProvider = AIProvider.Gemini;
+
+    const prompt = req.body.prompt;
+
+    try {
+      const AIConnection = genAIController.createConnection(AI);
+
+      const response = await AIConnection.generateResponse(prompt);
+
+      res.locals.response = response;
+
+      return next();
+    } catch (err: unknown) {
+      console.log('here');
```

Comment:
I'm a bit confused by the console logs and the github action warnings. Can we have an offline talk regarding them?

Reply:
This particular error (and some of the ones following them) crept in and should have been removed as I tried to resolve Linkta-196. I have removed these, but the GitHub action warnings still exist. We should have this conversation on a broader scope.